plugins/modules/network_settings_workflow_manager.py

```diff
@@ -873,60 +873,54 @@ def get_reserve_pool_params(self, pool_info):
             "name": pool_info.get("groupName"),
             "site_id": pool_info.get("siteId"),
         }
-        if len(pool_info.get("ipPools")) == 1:
+        pool_info_ippools = pool_info.get("ipPools")
+        if len(pool_info_ippools) == 1:
             reserve_pool.update({
-                "ipv4DhcpServers": pool_info.get("ipPools")[0].get("dhcpServerIps"),
-                "ipv4DnsServers": pool_info.get("ipPools")[0].get("dnsServerIps"),
+                "ipv4DhcpServers": pool_info_ippools[0].get("dhcpServerIps"),
+                "ipv4DnsServers": pool_info_ippools[0].get("dnsServerIps"),
                 "ipv6AddressSpace": "False"
             })
-            if pool_info.get("ipPools")[0].get("gateways") != []:
-                reserve_pool.update({"ipv4GateWay": pool_info.get("ipPools")[0].get("gateways")[0]})
+            if pool_info_ippools[0].get("gateways") != []:
+                reserve_pool.update({"ipv4GateWay": pool_info_ippools[0].get("gateways")[0]})
             else:
                 reserve_pool.update({"ipv4GateWay": ""})
             reserve_pool.update({"ipv6AddressSpace": "False"})
-        elif len(pool_info.get("ipPools")) == 2:
-            if not pool_info.get("ipPools")[0].get("ipv6"):
-                reserve_pool.update({
-                    "ipv4DhcpServers": pool_info.get("ipPools")[0].get("dhcpServerIps"),
-                    "ipv4DnsServers": pool_info.get("ipPools")[0].get("dnsServerIps"),
-                    "ipv6AddressSpace": "True",
-                    "ipv6DhcpServers": pool_info.get("ipPools")[1].get("dhcpServerIps"),
-                    "ipv6DnsServers": pool_info.get("ipPools")[1].get("dnsServerIps"),
-
-                })
-
-                if pool_info.get("ipPools")[0].get("gateways") != []:
-                    reserve_pool.update({"ipv4GateWay":
-                                        pool_info.get("ipPools")[0].get("gateways")[0]})
-                else:
-                    reserve_pool.update({"ipv4GateWay": ""})
+        elif len(pool_info_ippools) == 2:
+            ipv4_index = 0
+            ipv6_index = 0
+            if not pool_info_ippools[0].get("ipv6"):
+                ipv6_index += 1
+            elif not pool_info_ippools[1].get("ipv6"):
+                ipv4_index += 1
 
-                if pool_info.get("ipPools")[1].get("gateways") != []:
-                    reserve_pool.update({"ipv6GateWay":
-                                         pool_info.get("ipPools")[1].get("gateways")[0]})
-                else:
-                    reserve_pool.update({"ipv6GateWay": ""})
+            reserve_pool.update({
+                "ipv4DhcpServers": pool_info_ippools[ipv4_index].get("dhcpServerIps"),
+                "ipv4DnsServers": pool_info_ippools[ipv4_index].get("dnsServerIps"),
+                "ipv6AddressSpace": "True",
+                "ipv6Prefix": "True",
```

Comment:
We can either take prefix length or number of IP addresses. Can you please have a look and set it?